Lib/Inc/CThermistor.h

```diff
@@ -0,0 +1,28 @@
+/**
+ * @file CThermistor.h
+ */
+
+/*
+ *  Created on: Feb 15, 2022
+ *      Author: niuslar
+ */
+
+#ifndef CTHERMISTOR_H_
+#define CTHERMISTOR_H_
+
+#include "adc_data.h"
+
+class CThermistor
+{
+public:
+    CThermistor(CAdcData &adc_data, uint8_t adc_channel);
+
+    // Public methods
+    float getTemperature();
```

Comment:
This is somewhat weird, but it would be better if thermistor was a standalone class that took voltage and produced temperature. The reasons for it are:
1. Code reuse becomes easier if your class is not attached to an ADC class.
2. It is easiesr to test your code if you can feed it artificial voltage values for which you know what temperature should be returned. It allows for static testing of the code.
So I recommend this class is redone as follows:
```
CThermistor::CThermistor(float* p_calibration_array, uint8_t calibration_order)
{
    // calibration array contains a number of floats that participate in calibration of the thermistor
    // calibration order tells constructor how many variables are in the calibration array.
}